core/trino-spi/src/main/java/io/trino/spi/function/Signature.java

```diff
@@ -67,37 +70,36 @@ boolean isOperator()
         return name.startsWith(OPERATOR_PREFIX);
     }
 
-    @JsonProperty
     public String getName()
     {
         return name;
     }
 
-    @JsonProperty
     public TypeSignature getReturnType()
     {
         return returnType;
     }
 
-    @JsonProperty
+    public Optional<Function<List<TypeSignature>, TypeSignature>> getFunctionalReturnType()
```

Comment:
What's a "functional return type"?

Reply:
It's a method to allow building "return type" (e.g. `a + b`) from "function arguments" (e.g. `concat(a, b)`). The name might be confusing. Do you have any suggestion?

Reply:
`getReturnTypeDerivation` ?

Reply:
Thanks for your suggestion, changed to `getReturnTypeDerivation`.